fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/AbstractLoanRepaymentScheduleTransactionProcessor.java

```diff
@@ -205,6 +207,33 @@ public ChangedTransactionDetail reprocessLoanTransactions(final LocalDate disbur
         return changedTransactionDetail;
     }
 
+    protected void addChargeOnlyRepaymentInstallmentIfRequired(Set<LoanCharge> charges,
+            List<LoanRepaymentScheduleInstallment> installments) {
+        if (!CollectionUtils.isEmpty(charges) && !CollectionUtils.isEmpty(installments)) {
+            LoanRepaymentScheduleInstallment latestRepaymentScheduleInstalment = installments.get(installments.size() - 1);
+            LocalDate installmentDueDate = null;
+            for (final LoanCharge loanCharge : charges) {
```

Comment:
charges are a set, there is no order how you are iterating through on it, and it can happen multiple charges are after the last installment due date, no?

I would be happier if we were taking the latest charge which was after the maturity date and set that one instead of iterating though on them.